src/test/java/org/broadinstitute/hellbender/utils/clipping/ClippingOpUnitTest.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.hellbender.utils.clipping;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public final class ClippingOpUnitTest extends BaseTest {
+
+    @Test (dataProvider = "SoftClipedReadsNewStart")
+    public void testGetNewAlignmentStartOffset(final String preClip, final String postClip, final int result) {
+        Cigar cpreClip = TextCigarCodec.decode(preClip);
+        Cigar cpostClip = TextCigarCodec.decode(postClip);
+        Assert.assertEquals(ClippingOp.getNewAlignmentStartOffset(cpostClip, cpreClip), result,
+                "getNewAlignmentStartOffset returned "+ClippingOp.getNewAlignmentStartOffset(cpostClip, cpreClip)+
+                        " when "+result+" was expected for "+preClip.toString()+" which was clipped to "+postClip.toString());
+    }
+
+    @DataProvider(name = "SoftClipedReadsNewStart")
+    public Object[][] makeRevertSoftClipsBeforeContig() {
+        return new Object[][] {
+                {"70M", "10S60M", 10},
```

Comment:
Add a one-line comment at the top of your `DataProvider` explaining what each of the three values here mean.